sdk/core/Azure.Core/api/Azure.Core.net6.0.cs

```diff
@@ -236,6 +236,7 @@ public RequestFailedException(int status, string message) { }
         public RequestFailedException(int status, string message, System.Exception? innerException) { }
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
         public RequestFailedException(int status, string message, string? errorCode, System.Exception? innerException) { }
+        [System.ObsoleteAttribute(DiagnosticId="SYSLIB0051")]
```

Comment:
Curious - does this cause breaks anywhere?

Reply:
According to the docs it should not cause breaks:

> There might be cases where you can't remove these APIs from your custom exception type, for example, if you produce a library constrained by API compatibility requirements. In this case, the recommendation is to obsolete your own serialization constructor and GetObjectData methods using the SYSLIB0051 diagnostic code, as shown in the following code. Since ideally nobody outside the serialization infrastructure itself should be calling these APIs, obsoletion should only impact other types that subclass your custom exception type. It should not virally impact anybody catching, constructing, or otherwise using your custom exception type.

https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051#workaround